### ISSUE TITLE
.apk files don't work in pupy issue

### ISSUE THREAD

**Issue by AlphaMufasaOmega** [n1nj4sec/pupy#651]
so I have managed a successful install but i can't generate .apk files.. is it a problem with python for android? when it packages the .py file? could the problem be fixed if it were done manually? i keep getting the error code as shown below and want to know if anyone thinks that it is related.... let me know thanks!
```
[+] loading scriptlet 'hide_argv' with args {'name': 'architect'}
[+] loading scriptlet 'keylogger' with args {}
[+] loading scriptlet 'daemonize' with args {}
[+] loading scriptlet 'persistence' with args {'method': 'registry'}
[+] Generate client: android/x86
[C] launcher: connect
[C] launcher_args: ['-t', 'ssl', '--host', 'IPCENSORED:443']
[C] cid: 0xea3c72bfL
[C] debug: False
[+] Required credentials (found):
[+] SSL_BIND_CERT, SSL_CA_CERT, SSL_CLIENT_CERT, SSL_BIND_KEY, SSL_CLIENT_KEY
[+] generating PY payload ...
[+] packaging the apk ... (can take 10-20 seconds)
/bin/sh: 1: jarsigner: not found
[-] payload generation failed: Command 'jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore crypto/pupy-apk-release-key.keystore -storepass pupyp4ssword '/tmp/tmp_pupy_GoLK14' pupy_key' returned non-zero exit status 127

```

**Comment by alxchk** [comment]
> is it a problem with python for android?

It's problem with absense of `jarsigner`.

```
> dpkg -S jarsigner                                                                                                                                                                                          
bash-completion: /usr/share/bash-completion/completions/jarsigner
openjdk-8-jdk-headless:i386: /usr/lib/jvm/java-8-openjdk-i386/man/man1/jarsigner.1.gz
openjdk-8-jdk-headless:i386: /usr/lib/jvm/java-8-openjdk-i386/bin/jarsigner
openjdk-8-jdk-headless:i386: /usr/lib/jvm/java-8-openjdk-i386/man/ja_JP.UTF-8/man1/jarsigner.1.gz
```

```
> ls -l /usr/bin/jarsigner
lrwxrwxrwx 1 root 27 Jan 13  2017 /usr/bin/jarsigner -> /etc/alternatives/jarsigner
```

**Comment by Strazzom** [comment]
Did you try installing openjdk as @alxchk recommended over at #639?

**Comment by AlphaMufasaOmega** [comment]
yes, after installing open jdk with `sudo apt-get install default-jdk`Command 'jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore crypto/pupy-apk-release-key.keystore -storepass pupyp4ssword '/tmp/tmp_pupy_E4cTkT' pupy_key' returned non-zero exit status 1

**Comment by SammySet007** [comment]
I have the same issue (i.e jarsigner is installed but error code 1) #665 I have been advised its due to the pupy-apk-release-key.keystore file being the issue. 

If I run 
`jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore crypto/pupy-apk-release-key.keystore -storepass pupyp4ssword '/tmp/tmp_pupy_E4cTkT' pupy_key`

The temp file will be missing (even based on my own temp file) so not sure where to test from here. A docker instance works fine however I need this to work from a public IP so the docker instance becomes an issue here (limited docker knowledge)

**Comment by SammySet007** [comment]
Also within pupysh if i generate an apk I also get the error

`payload generation failed: Command 'jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore crypto/pupy-apk-release-key.keystore -storepass pupyp4ssword '/tmp/tmp_pupy_gSaL_C' pupy_key' returned non-zero exit status 1
`

**Comment by Strazzom** [comment]
If you are testing this on Debian 9, this may be a dependency issue.

The dependencies listed in the wiki are not the complete set required to run pupy in all circumstances, (as jarsigner shows pretty clearly).

Look at the dockerfile in the pupy directory, (where pupysh is). One of the lines in there lists all the dependencies that the docker image installs. Try installing all of them, and then report back.

**Comment by Strazzom** [comment]
If you are testing this on Debian 9, this may be a dependency issue.

The dependencies listed in the wiki are not the complete set required to run pupy in all circumstances, (as jarsigner shows pretty clearly).

Look at the dockerfile in the pupy directory, (where pupysh is). One of the lines in there lists all the dependencies that the docker image installs. Try installing all of them, and then report back.
